one-ui/ui/src/components/entities/mapping/mapping-card.tsx

```diff
@@ -177,83 +185,233 @@ const MappingCard: React.FC<Props> = (props) => {
 
     }
 
-    const fetchSrcDocFromUri = async (uri) => {
+    const fetchSrcDocFromUri = async (uri, index = mapIndexLocal) => {
         try{
             let srcDocResp = await getDoc('STAGING', uri);
             if (srcDocResp.status === 200) {
+                let parsedDoc: any;
+                if(typeof(srcDocResp.data) === 'string'){
+                    parsedDoc = getParsedXMLDoc(srcDocResp);
+                } else {
+                    parsedDoc = srcDocResp.data;
+                }
+                if(parsedDoc['envelope']){
+                    if(parsedDoc['envelope'].hasOwnProperty('@xmlns')){
+                        
+                        let nmspcURI = parsedDoc['envelope']['@xmlns']
+                        let indCheck = nmspcURI.lastIndexOf('/');
+                        let ind = indCheck !== -1 ? indCheck + 1 : 0;
+                        let nmspcString = nmspcURI.slice(ind);
+                        namespaceString = nmspcString;
+                        nmspaces = { ...nmspaces, [namespaceString]: nmspcURI};
+                        setNamespaces({ ...namespaces, [namespaceString]: nmspcURI})
```

Comment:
Could we have just used `namespaceString = nmspcURI.slice(ind);`?
Also looks like we are trying to get a namespace string in 4 to 5 different places similar to line 201 to 205. Can we create a function and reuse that to return the namespace string.

Reply:
This logic to set namespaceString is used only at this very place so seems an overkill to create a function for only one instance.

Reply:
nmspcString is extracted only at two places, one is inside the function updateParentNamespace() (already being called at multiple places) and other at line 204-205. The reason why that function is not called at these lines is that the concept of "parentNamespace" didn't made sense at this point earlier but only once you enter into the nested source logic. If still however we want to replace that code with a function call, then I will have to rename the function to something else which fits both the scenarios.

Reply:
Discussed with @xnikhil08 . Will resolve this conversation and refactor the code later when writing tests or as a tech debt.